backend/src/middleware/validation/index.js

```diff
@@ -1,16 +1,5 @@
 import { UserInputError } from 'apollo-server'
 
-const USERNAME_MIN_LENGTH = 3
-
-const validateUsername = async (resolve, root, args, context, info) => {
-  if (!('name' in args) || (args.name && args.name.length >= USERNAME_MIN_LENGTH)) {
-    /* eslint-disable-next-line no-return-await */
-    return await resolve(root, args, context, info)
-  } else {
-    throw new UserInputError(`Username must be at least ${USERNAME_MIN_LENGTH} characters long!`)
-  }
-}
-
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T16:14:14Z" title="Wednesday, July 3rd 2019, 6:14:14 pm +02:00">Jul 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,16 +1,5 @@
 import { UserInputError } from 'apollo-server'
 
-const USERNAME_MIN_LENGTH = 3
-
-const validateUsername = async (resolve, root, args, context, info) => {
-  if (!('name' in args) || (args.name && args.name.length >= USERNAME_MIN_LENGTH)) {
-    /* eslint-disable-next-line no-return-await */
-    return await resolve(root, args, context, info)
-  } else {
-    throw new UserInputError(`Username must be at least ${USERNAME_MIN_LENGTH} characters long!`)
-  }
-}
-
```


We are not using usernames any longer?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:01:41Z" title="Wednesday, July 3rd 2019, 10:01:41 pm +02:00">Jul 3, 2019</time>_

---


No, this is basically a refactoring of the existing code base. For some strange reason we want to force our users to type in a 3-letter minimum username.

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-04T06:29:58Z" title="Thursday, July 4th 2019, 8:29:58 am +02:00">Jul 4, 2019</time>_

---


So we allow one character usernames yet?